bundle/edu.gemini.qv.plugin/src/main/scala/edu/gemini/qv/plugin/QvContext.scala

```diff
@@ -45,11 +46,11 @@ case class QvContext(peer: Peer, dataSource: DataSource, source: ObservationProv
   def timezone = site.timezone()
 
   // ===== Data providers / filters used in this context ====
-  private val _nonSidProvider = new PositionProvider(this, source)
-  private val _mainFilter = FilterProvider(_nonSidProvider)
-  private val _tableFilter = FilterProvider(_mainFilter)
-  private val _selection = SelectionProvider(_tableFilter)
-  
+  private lazy val _nonSidProvider = new PositionProvider(this, source)
+  private lazy val _mainFilter = FilterProvider(this, _nonSidProvider)
+  private lazy val _tableFilter = FilterProvider(this, _mainFilter)
+  private lazy val _selection = SelectionProvider(this, _tableFilter)
```

Comment:
I admit that I don't have a well founded argument against it but I think [this SIP entry](http://docs.scala-lang.org/sips/pending/improved-lazy-val-initialization.html) does a good job of indicating the complexity and potential for deadlock.  Locking and checking the initialization state is involved as well with every access to a `lazy val`, so if it is heavily used then it isn't clear you've gained anything.  Unless there's something that is really expensive to initialize and yet not always needed, or impossible to initialize on construction it just seems like a premature optimization.